scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/executors/forked/env/ExecuteForkedTaskInsideNewJvm.java

```diff
@@ -0,0 +1,100 @@
+ * modify it under the terms of the GNU Affero General Public License
+ * as published by the Free Software Foundation; version 3 of
+ * the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this library; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307
+ * USA
+ *
+ * If needed, contact us to obtain a release under GPL Version 2 or 3
+ * or a different license than the AGPL.
+ *
+ *  Initial developer(s):               The ProActive Team
+ *                        http://proactive.inria.fr/team_members.htm
+ *  Contributor(s):
+ *
+ *  * $$ACTIVEEON_INITIAL_DEV$$
+ */
+package org.ow2.proactive.scheduler.task.executors.forked.env;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+import org.ow2.proactive.scheduler.task.TaskResultImpl;
+import org.ow2.proactive.scheduler.task.context.TaskContext;
+import org.ow2.proactive.scheduler.task.executors.InProcessTaskExecutor;
+import org.apache.commons.io.FileUtils;
+
+public class ExecuteForkedTaskInsideNewJvm {
+
+    // 2 called by forked
+    private static TaskContext deserializeContext(String pathToFile) throws IOException,
+            ClassNotFoundException {
+        File f = new File(pathToFile);
+        try (ObjectInputStream inputStream = new ObjectInputStream(new FileInputStream(f))) {
+            return (TaskContext) inputStream.readObject();
+        } finally {
+            FileUtils.forceDelete(f);
+        }
+    }
+
+    // 3 called by forked
+    private static void serializeTaskResult(Object result, String contextPath) throws IOException {
+        File file = new File(contextPath);
+        try (ObjectOutputStream objectOutputStream = new ObjectOutputStream(new FileOutputStream(file))) {
+            objectOutputStream.writeObject(result);
+        }
+    }
+
+    /**
+     * Everything here and called from here should only use System.out and System.err
+     */
+    public static void main(String[] args) throws Throwable {
+        if (args.length != 1) {
+            System.err.println("Path to serialized task context is expected");
+            System.exit(-1);
+        }
+
+        fromForkedJVM(args[0]);
+    }
+
+    private static void fromForkedJVM(String contextPath) {
+        try {
+            TaskContext container = deserializeContext(contextPath);
+
+            TaskResultImpl result = new InProcessTaskExecutor().execute(container, System.out, System.err);
+
+            serializeTaskResult(result, contextPath);
+        } catch (Throwable throwable) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1181)